services/wallet/collectibles/manager.go

```diff
@@ -1027,3 +1029,93 @@ func (o *Manager) SearchCollections(ctx context.Context, chainID walletCommon.Ch
 	}
 	return nil, ErrNoProvidersAvailableForChainID
 }
+
+func (o *Manager) FetchCollectibleSocialsAsync(ctx context.Context, uniqueID thirdparty.CollectibleUniqueID) error {
+	hasValidSocials := o.collectionsDataDB.GetSocialsValidForID(uniqueID)
+
+	if !hasValidSocials {
+		// Atomic group stores the error from the first failed command and stops other commands on error
+		group := async.NewAtomicGroup(ctx)
+		group.Add(func(ctx context.Context) error {
+			defer o.checkConnectionStatus(uniqueID.ContractID.ChainID)
+
+			socials, err := o.fetchSocialsForCollectibleUniqueID(ctx, uniqueID.ContractID.ChainID, uniqueID)
+			if err != nil {
+				log.Debug("FetchCollectibleSocials failed for", "chainID", uniqueID.ContractID.ChainID, "uniqueID", uniqueID, "err", err)
+				return err
+			}
+
+			socialsMessage := CollectibleSocialsMessage{
+				ID:      uniqueID,
+				Socials: socials,
+			}
+
+			err = o.collectionsDataDB.SetCollectionSocialsData(uniqueID.ContractID, socials)
+			if err != nil {
+				log.Error("Error saving socials to DB: %v", err)
+				return nil
+			}
+
+			payload, err := json.Marshal(socialsMessage)
+			if err != nil {
+				log.Error("Error marshaling response: %v", err)
+				return nil
+			}
+
+			event := walletevent.Event{
+				Type:    EventGetCollectibleSocialsDone,
+				Message: string(payload),
+			}
+
+			o.feed.Send(event)
+			return err
+		})
+
+		group.Wait()
+		return group.Error()
+	}
+	return nil
+}
+
+func (o *Manager) fetchSocialsForCollectibleUniqueID(ctx context.Context, chainID walletCommon.ChainID, idToFetch thirdparty.CollectibleUniqueID) (thirdparty.CollectionSocials, error) {
+	socials := thirdparty.CollectionSocials{}
+	providerFound := false
+	cmdRes := circuitbreaker.CommandResult{}
+	for _, provider := range o.providers.CollectibleDataProviders {
+		if provider.ID() == alchemy.AlchemyID {
```

Comment:
Don't do specific provider checks here, have whichever provider can't provide the required info return `ErrEndpointNotSupported` instead